src/core/create-higlass-models.ts

```diff
@@ -68,5 +68,23 @@ export function renderHiGlass(
             });
     });
 
-    setHg(hgModel.spec(), getBoundingBox(trackInfos), spec);
+    const trackInfosWithShapes: TrackMouseEventData[] = trackInfos.map(d => {
+        return {
+            id: d.track.id!,
+            spec: d.track as SingleTrack | OverlaidTrack,
+            shape:
+                d.track.layout === 'linear'
+                    ? d.boundingBox
+                    : {
+                          cx: d.boundingBox.x + d.boundingBox.width / 2.0,
+                          cy: d.boundingBox.y + d.boundingBox.height / 2.0,
+                          innerRadius: d.track.innerRadius!,
+                          outerRadius: d.track.outerRadius!,
+                          startAngle: d.track.startAngle!,
+                          endAngle: d.track.endAngle!
```

Comment:
Will these fields always be defined? if not, we shouldn't tell the compiler they are and instead model the potential missing-ness in the `TrackMouseEventData` type.

Reply:
Yes, it is. It is defined as the very first step of compiling (similar to https://github.com/gosling-lang/gosling.js/pull/779#discussion_r933222279). But, it would be best if we could figure out a nice way to define a separate type that we can internally use after compiling (e.g., making these properties required).